source/common/router/upstream_request.h

```diff
@@ -30,8 +31,52 @@ class GenericUpstream;
 class GenericConnectionPoolCallbacks;
 class RouterFilterInterface;
 class UpstreamRequest;
+class UpstreamRequestFilterManagerCallbacks;
+class UpstreamFilterManager;
+class CodecFilter;
 
-// The base request for Upstream.
+/* The Upstream request is the base class for forwarding HTTP upstream.
```

Comment:
What an awesome comment! Thanks for writing it.

Reply:
it'd be kinda cruel to make you figure it out on your own :-)